packages/core/src/Engine.ts

```diff
@@ -300,10 +309,15 @@ export class Engine extends EventDispatcher {
       componentsManager.callAnimationUpdate(deltaTime);
       componentsManager.callScriptOnLateUpdate(deltaTime);
       this._render(scene);
-      componentsManager.handlingInvalidScripts();
     }
 
     engineFeatureManager.callFeatureMethod(this, "postTick", [this, this._sceneManager._activeScene]);
+
+    // Engine is complete delayed destruction mechanism
+    if (this._waittingDestroy) {
+      this._destroy();
+    }
+    componentsManager.handlingInvalidScripts();
```

Comment:
Why not put it before destroy?

Reply:
That not wor,  `this._sceneManager._destroy();`  is in `this._destroy()`

Reply:
For example:
```
onDestroy(): void {
     console.log('Destroyed at', this.engine.time.nowTime());
}
```

Reply:
Understood.

Reply:
done